### ISSUE TITLE
Several inlet tests are failing on Windows: int32 vs. int64

### ISSUE THREAD

**Issue by kennyweiss** [LLNL/axom#489]
PR #488 fixes some compiler issues that prevented inlet from building on Windows.

Now that inlet is building on Windows, there are several failing tests:
* inlet_nested_structs_ex
* inlet_object
* inlet_Inlet

They all appear to be failing due to differences in interpretation of `long` as `int64` (expected) vs. `int32` (actual)
E.g., I got the following error log:
```
145: ***********************************
145: [WARNING in line 14253 of file D:\Projects\axom\uberenv_libs\vcpkg\buildtrees\conduit\src\v0.6.0-e451019a30\src\libs\conduit\conduit_node.cpp]
145: MESSAGE=Node::as_long_array() const -- DataType int64 at path arr does not equal expected DataType int32
145: ***********************************
145: D:\Projects\axom\src\axom\inlet\tests\inlet_object.cpp(874): error: Expected equality of these values:
145:   arr
145:     Which is: {}
145:   expected_arr
145:     Which is: { 4, 6, 10 }
145: D:\Projects\axom\src\axom\inlet\tests\inlet_object.cpp(876): error: Expected equality of these values:
145:   arr
145:     Which is: {}
145:   expected_arr
145:     Which is: { 4, 6, 10 }
145: D:\Projects\axom\src\axom\inlet\tests\inlet_object.cpp(881): error: Expected equality of these values:
145:   arr_w_indices
145:     Which is: {}
145:   expected_arr_w_indices
145:     Which is: { (0, 4), (1, 6), (2, 10) }
145: [  FAILED  ] inlet_object/1.primitive_arrays_as_std_vector, where TypeParam = class axom::inlet::JSONReader (5 ms)
145: [ RUN      ] inlet_object/1.primitive_arrays_as_std_vector_wrong_type
```

Hopefully these can be resolved via a configuration change?

Tag: @cyrush

**Comment by joshessman-llnl** [comment]
Thanks for looking into this!

Without seeing the stacktrace I can't be completely sure but if this is from 
https://github.com/LLNL/axom/blob/0e71819fd131cad332b843796ea3a0244e9b9764/src/axom/inlet/ConduitReader.cpp#L442

it may be sufficient to just change `as_long_array` to `as_int64_array`.

**Comment by kennyweiss** [comment]
Resolved in #502
